eslint/babel-eslint-parser/src/convert/convertTokens.js

```diff
@@ -173,6 +173,8 @@ function convertToken(token, source) {
   } else if (type === tt.bigint) {
     token.type = "Numeric";
     token.value = `${token.value}n`;
+  } else if (type === tt.privateName) {
+    token.type = "PrivateIdentifier";
```

Comment:
We convert `tt.privateName` to `PrivateIdentifier`, which aligns to https://github.com/eslint/espree/pull/486

Note that although this PR merges `tt.hash` and `tt.name` to `tt.privateName`, this behaviour will be not observed by `@babel/eslint-parsers` because of the compat layer. However if we run with `BABEL_8_BREAKING=true`, the eslint parser will see `tt.privateName`, instead of breaking `tt.privateName`, we align it to the new espree behaviour.